src/core/load-dropin.c

```diff
@@ -25,10 +25,14 @@ static int process_deps(Unit *u, UnitDependency dependency, const char *dir_suff
                 return r;
 
         STRV_FOREACH(p, paths) {
-                _cleanup_free_ char *target = NULL;
-                const char *entry;
+                _cleanup_free_ char *target = NULL, *target_file = NULL, *entry = NULL;
 
-                entry = basename(*p);
+                r = path_extract_filename(*p, &entry);
+                if (r < 0) {
+                        log_unit_warning_errno(u, r, "%s couldn't read basename of dropin %s, ignoring: %m",
```

Comment:
The error message sounds strange: e.g. `Wants couldn't read basename of ...` ???